binlog_streamer.go

```diff
@@ -184,6 +188,7 @@ func (s *BinlogStreamer) Run() {
 			// so there's no way to handle this for us.
 			continue
 		default:
+			s.lastResumableBinlogPosition.Pos = uint32(ev.Header.LogPos)
```

Comment:
This is making some assumption about what's the `default:` case 🤔 IIRC, you said we wanna record XIDEvent..

should that be made explicit?

Reply:
Yes. 

Following up on our discussion, we see the `GTIDEvent` (if enabled) at the start of a transaction, and an `XIDEvent` at the end of a transaction. 

I've updated this block to reflect that and have also renamed the state tracker's methods to use `Resumable` in place of `Written` for the `BinlogWriter` and `InlineVerifier`'s resumable positions 👍